src/test/java/uk/gov/pay/ledger/transactionsummary/dao/TransactionSummaryDaoIT.java

```diff
@@ -145,4 +154,70 @@ public void shouldUpdateFeeForTransactionSummaryCorrectly() {
         assertThat(transactionSummary.get(0).get("total_amount_in_pence"), is(1000L));
         assertThat(transactionSummary.get(0).get("total_fee_in_pence"), is(123L));
     }
+
+    @Test
+    public void verifyMonthlyGatewayPerformanceReportTest() {
+        aTransactionSummaryFixture()
+                .withGatewayAccountId("1")
+                .withAmount(1000L)
+                .withState(STARTED)
+                .withType(PAYMENT)
+                .withLive(true)
+                .withTransactionDate(ZonedDateTime.parse("2019-01-01T02:00:00Z"))
+                .insert(rule.getJdbi());
+
+        aTransactionSummaryFixture()
+                .withGatewayAccountId("1")
+                .withAmount(1000L)
+                .withState(SUCCESS)
+                .withType(REFUND)
+                .withLive(true)
+                .withTransactionDate(ZonedDateTime.parse("2019-01-01T02:00:00Z"))
+                .insert(rule.getJdbi());
+
+        aTransactionSummaryFixture()
+                .withGatewayAccountId("1")
+                .withAmount(1000L)
+                .withState(SUCCESS)
+                .withType(PAYMENT)
+                .withLive(true)
+                .withTransactionDate(ZonedDateTime.parse("2018-01-01T02:00:00Z"))
+                .insert(rule.getJdbi());
+
+        List<String> relevantGatewayAccounts = List.of("1", "2");
+        relevantGatewayAccounts.forEach(account -> aTransactionSummaryFixture()
+                .withGatewayAccountId(account)
+                .withAmount(1000L)
+                .withNoOfTransactions(1L)
+                .withState(SUCCESS)
+                .withType(PAYMENT)
+                .withLive(true)
+                .withTransactionDate(ZonedDateTime.parse("2019-01-01T02:00:00Z"))
+                .insert(rule.getJdbi()));
+
+        BigDecimal expectedValue = BigDecimal.valueOf(1000);
+        LocalDate startDate = LocalDate.parse("2019-01-01");
+        LocalDate endDate = LocalDate.parse("2019-02-01");
+
+        List<GatewayAccountMonthlyPerformanceReportEntity> performanceReport = transactionSummaryDao.monthlyPerformanceReportForGatewayAccounts(startDate, endDate);
+
+        GatewayAccountMonthlyPerformanceReportEntity gatewayAccountOnePerformanceReport = performanceReport.get(0);
+        GatewayAccountMonthlyPerformanceReportEntity gatewayAccountTwoPerformanceReport = performanceReport.get(1);
+
+        assertThat(performanceReport.size(), CoreMatchers.is(2));
+
+        assertThat(gatewayAccountOnePerformanceReport.getGatewayAccountId(), CoreMatchers.is(1L));
+        assertThat(gatewayAccountOnePerformanceReport.getTotalVolume(), CoreMatchers.is(1L));
+        assertThat(gatewayAccountOnePerformanceReport.getTotalAmount(), CoreMatchers.is(closeTo(expectedValue, ZERO)));
```

Comment:
is there any reason `closeTo` is used to assert rather than exact match ?

Reply:
Ah I copied and pasted these tests from PerformanceReportDao. Yes `closeTo` shouldn't be used with getTotalAmount but it has to be used for getAverageAmount as it returns 1000.000... and the expected value is 1000. Since both are BigDecimal they won't be equal so I had to use `closeTo`